week-04-associated-data-and-users/day-02/module-01/readme.md

```diff
@@ -0,0 +1,331 @@
+
+
+
+Now we will create a `Console` document using the `nin64` object we made above.  While inside the console creation callback function, we'll also create our `Game` document.  We do this inside the `Console` creation because we can easily access the newly created `nintendo64` object this is available to us as a callback return in line 3.  
+
+After we create the `Game` document, we push the `nintendo64` console document into the `zeldaGame` consoles array.  Since we already told the Game Schema that we will only be storing ObjectIds instead of actual `Console` documents in the `consoles` array, mongoose will convert `nintendo64` to it's unique `_id` .
+
+```javascript
+ Console.create(nin64, function(err, nintendo64){
+ 	if(err) {return console.log(err);}
+ 	Game.create(zelda, function(err, zeldaGame) {
+ 		if(err) {return console.log(err);}
+ 		zeldaGame.consoles.push(nintendo64);
+ 		zeldaGame.save();
+ 		console.log('Game success: \n' + zeldaGame);
+ 	});
+ })
+```
+This is the result after executing the code we've written thus far:
+
+```javascript
+{ __v: 0,
+  name: 'The Legend of Zelda: Ocarina of Time',
+  developer: 'Nintendo',
+  _id: 55e4eb857d6157f4d41a2981,
+  consoles: [ 55e4eb857d6157f4d41a2980 ] }
+
+```
+What are we looking at?
+	- Line 1: __v represents the number of times the document has been accessed
+	- Line 2: The name property of the Game Document we have created.
+	- Line 3: The developer property of the Game Document we have created.
+	- Line 4: The unique `_id` created by Mongoose for our Game Document.
+	- Line 5: The consoles array with a single `ObjectId` that is associated with our Console Document.
+
+Lets print out the Console Document `nintendo64` to make sure the `ObjectId` in consoles matches the `_id`:
+
+```javascript
+
+{ _id: 55e4eb857d6157f4d41a2980,
+    name: 'Nintendo 64',
+    manufacturer: 'Nintendo',
+    released: Sun Sep 29 1996 00:00:00 GMT-0700 (PDT),
+    __v: 0 }
+
+```
+
+Sure enough, the Game Document consoles only `ObjectId` matches the Console Document `_id`.  What's going on?  The Game Document consoles has a single `Objectid` that contians the '*address*' or the '*location*' where it can find the Console Document if and when it needs it.  This keeps our Game Document small, since it doesn't have to have so much information packed into it.  When it wants the Console Document data, it will ask for it. Until then, it's happy with just the `ObjectId` associated with it.
+
+##The `populate()` method
+
+When we want to finally get information from any Console Document we have inside the Game Document consoles array, we use the method call ```.populate()```.  
+
+```javascript
+Game.findOne({ name: 'The Legend of Zelda: Ocarina of Time' })
+	.populate('consoles')
+	.exec(function(err, game) {
+ 		if(err){return console.log(err);}
+ 		console.log(game);
+ 		console.log("/nI love " + game.name + " for the " + game.consoles[0].name);
+ 	});
+```
```

Comment:
@eerwitt Just double checking because there's a lot of repetition - are these supposed to be all of the same examples from the day before (day 1 - module 2)?